Add a fail test for accounts_password_all_shadowed

jan-cerny opened the pull request:
#### Description:
This rule missed a fail test scenario. We will add a fake user
with a random string different than x or asterisk in the second
field in /etc/passwd.

#### Rationale:

This PR will improve the test coverage of RHEL 7 CIS profiles.

marcusburghardt reviewed (changes requested):
@jan-cerny , the test is fine. However, we should also update the OVAL here. For example, this line would also match as valid:
xtestuser:0fcf3792f5fffc4fcd1801fab37e0470c66d46cb0f01:8000:8000:testuser:/home/testuser:/bin/bash

In this case, the password is not shadowed but the test would pass because the username contains a x.

marcusburghardt commented:
> @jan-cerny , the test is fine. However, we should also update the OVAL here. For example, this line would also match as valid:
> xtestuser:0fcf3792f5fffc4fcd1801fab37e0470c66d46cb0f01:8000:8000:testuser:/home/testuser:/bin/bash
> 
> In this case, the password is not shadowed but the test would pass because the username contains a x.

This should solve:
`:x:|:\*:`

jan-cerny commented:
@marcusburghardt Amazing! I have changed the OVAL and added another test scenario. But I think the regex needs to be different than you propose because the OVAL `password` element contains only the password value , ie. without colons.

marcusburghardt reviewed (approved):
Great. Thanks @jan-cerny
